### ISSUE TITLE
no-undef Promise

### ISSUE THREAD

**Issue by dermike** [zaggino/brackets-eslint#51]
Hello!

Thanks for this great plugin!
I have a, for me at least, weird problem with it. In Brackets with brackets-eslint I get this error:

`ERROR: 'Promise' is not defined. [no-undef]	return new Promise((resolve, reject) => {`

I don't get the error however when checking it with eslint manually. As far as I can tell I'm using eslint 3.12.0 both globally and in brackets-eslint. My `.eslintrc` is in my `~/`directory.

I can get around the error by adding `/* global Promise: true */`, but wish I didn't have to since it works without it in "normal" eslint. Hope this can be fixed or if anyone can spot something I've setup wrong.

cheers!

**Comment by zaggino** [comment]
Hi @dermike 
have you tried this?
https://github.com/eslint/eslint/issues/4085#issuecomment-220863125

**Comment by dermike** [comment]
No error with gulp-eslint either, only brackets-eslint

**Comment by zaggino** [comment]
If you can create a sample project I can clone and replicate the issue I'm happy to take a look. Otherwise I won't be guessing. I use this daily for a large number of projects without issue, although I must admit, all of them have `.eslintrc` in their project root. I have never tried using `.eslintrc` in home directory.

**Comment by zaggino** [comment]
What's your OS, Mac or Linux? I'm primary a Windows user, not sure if the node domain of Brackets doesn't run under a different user so it can't find the `.eslintrc` in home dir.

**Comment by dermike** [comment]
I'm on Mac OS. I've created this repo which also includes my `.eslintrc`:
https://github.com/dermike/brackets-eslint-issue

Still getting the no-undef error with brackets-eslint even if it's in the project dir.

**Comment by zaggino** [comment]
If you have `.eslintrc` in your project, brackets-eslint requires you to have installed `eslint` in your project dir and uses that installation. It only uses global installation when there's no `.eslintrc` defined. Rationale is that if you specify `.eslintrc` file, you also need to install exact version of `eslint` you want to lint with.

When I do `npm install eslint` in your project after cloning it my result looks like this:
![image](https://cloud.githubusercontent.com/assets/1067319/21113195/b4a3647a-c0fc-11e6-909a-9094943bf90b.png)

Maybe you have an old version of `eslint` inside your project's `node_modules` ?

**Comment by dermike** [comment]
No, I only have it installed globally... installed latest version in the repo now and still the same error. Must be something Mac specific then I guess.

**Comment by zaggino** [comment]
Have you tried re-installing `brackets-eslint` ?
Latest version should ask you to install `eslint` locally:
https://github.com/zaggino/brackets-eslint/blob/7cf8f61a73edd4c9f1f6d3bb64c71be5784df39d/src/node/eslint.ts#L279-L283

**Comment by dermike** [comment]
Remove and reinstalled and same error... did not ask me to install locally.

**Comment by zaggino** [comment]
What version of Brackets are you on? 1.7 or 1.8? I'll check the code why that's the case.

**Comment by zaggino** [comment]
Try updating to `brackets-eslint@3.0.3`. I can't replicate this so I won't spend any more time on this but feel free to dig around if you want and submit a PR.

**Comment by dermike** [comment]
Tried with 3.0.3 and it works if I open the project directory in Brackets. Both the undef and ask to install eslint in the project. So this probably worked yesterday too, only had the file from the project open, not the project itself. So try reproducing it if you have another project open, and just open `test.js` separately.

Still getting the error though if I don't have a project local `.eslintrc` and eslint installed, or the current project dir opened. (I usually don't switch projects completely when I open other files from other projects). It works on other files outside of the current project as long as eslint is installed in there.

Wish there was a way it worked with my home directory `.eslintrc` only, but adding that globals config is good enough. Thanks!
